packages/ui-kit/src/components/PieChart/PieChart.tsx

```diff
@@ -0,0 +1,326 @@
+      leaderboardData?.leaderboard?.rows.push(['Other', otherValue.toString()])
+
+      return leaderboardData
+    }, [leaderboardData, counterData])
+
+    const loadingStyles = {
+      opacity: isLoading || isLoadingStatic ? '0.3' : '1',
+      transition: 'opacity 0.2s ease-in-out'
+    }
+
+    // @TODO: we should abstract this logic to a hook
+    React.useEffect(() => {
+      function handlePropsMismatch() {
+        if (isStatic && !headers && !rows) {
+          // console.error('InvalidPropsError: You must pass either `headers` and `rows` or `query` props') we will set logs as a feature later
+          setPropsMismatch(true)
+          return
+        }
+
+        if (isStatic && (!headers || !rows)) {
+          // console.error('InvalidPropsError: When passing the data via props you must pass both `headers` and `rows`') we will set logs as a feature later
+          setPropsMismatch(true)
+
+          return
+        }
+
+        if (
+          !isStatic &&
+          (hasError?.name === 'AccessTokenError' ||
+            !query.metric ||
+            !query.timeRange ||
+            !query.dimensions ||
+            !query.rowLimit)
+        ) {
+          // console.error(
+          //   'InvalidPropsError: When opting for fetching data you must pass at least `accessToken`, `metric`, `dimensions`, `rowLimit` and `timeRange` in the `query` prop'
+          // ) we will set logs as a feature later
+          setPropsMismatch(true)
+          return
+        }
+
+        if (variant !== 'pie' && variant !== 'doughnut') {
+          // console.error('InvalidPropsError: `variant` prop must be either `pie` or `doughnut`') we will set logs as a feature later
+          setPropsMismatch(false)
+        }
+
+        setPropsMismatch(false)
+      }
+
+      if (!isLoadingStatic) {
+        handlePropsMismatch()
+      }
+    }, [isStatic, headers, rows, query, isLoadingStatic, variant, hasError?.name])
+
+    React.useEffect(() => {
+      if (isStatic) {
+        renderChart({ headers, rows })
+      }
+    }, [isStatic, isLoadingStatic, variant, headers, rows, renderChart])
+
+    React.useEffect(() => {
+      if (fetchedData?.leaderboard && !isStatic) {
+        renderChart(fetchedData.leaderboard)
+      }
+    }, [fetchedData, variant, isStatic, renderChart])
+
+    React.useEffect(() => {
+      return () => {
+        destroyChart()
+      }
+    }, [])
+
+    if (hasError || propsMismatch) {
+      destroyChart()
+      return <ErrorFallback {...errorFallbackProps} error={error} />
+    }
+
+    const isNoContainerRef = (variant === 'pie' || variant === 'doughnut') && !canvasRef.current
+
+    if (((isStatic && isLoadingStatic) || (!isStatic && isLoading)) && isNoContainerRef) {
+      destroyChart()
+      return <Loader {...loaderProps} />
+    }
+
+    const isTotal = isStatic
+      ? (rows?.map((e) => Number(e[1])) ?? []).reduce((a: number, b: number) => a + b, 0) ?? '0'
+      : Number(counterData?.counter?.value)
```

Comment:
This line turns very difficult to read, I would suggest abstracting this to an util function, using more meaningful names and not doing everything in a single line.

Additionally I suggest using a different name than `isTotal` since the `is` prefix is commonly used for `boolean` variables but this one is a number, just `total` should be fine.